modules/openapi-generator/src/main/resources/php/ObjectSerializer.mustache

```diff
@@ -301,17 +325,26 @@ class ObjectSerializer
                 }
             }
             $instance = new $class();
-            foreach ($instance::openAPITypes() as $property => $type) {
-                $propertySetter = $instance::setters()[$property];
-
-                if (!isset($propertySetter) || !isset($data->{$instance::attributeMap()[$property]})) {
-                    continue;
+            $reverseAttributes = array_flip($instance::attributeMap());
+            if (!is_object($data)) {
+                return null;
+            }
+            if (!empty($instance::oneOf())) {
+                return self::deserialize($data, "oneOf[". implode(', ', $instance->onefOf())  ."]", $httpHeaders);
+            }
+            foreach(get_object_vars($data) as $key => $propertyValue) {
```

Comment:
we need to go through all the data fields and and see if those are settable to the current model instead of looking at the setters so that we can cull the oneof options. This cause us to also require a mechanism for setting additional properties